com.unity.netcode.gameobjects/Components/HalfVector3AxisToSynchronize.cs

```diff
@@ -0,0 +1,41 @@
+
+namespace Unity.Netcode.Components
+{
+    /// <summary>
+    /// Structure that defines which axis of a <see cref="HalfVector3"/>
+    /// and <see cref="HalfVector3DeltaPosition"/> will be serialized.
+    /// </summary>
+    public struct HalfVector3AxisToSynchronize
+    {
+        /// <summary>
+        /// When enabled, serialize the X axis
+        /// </summary>
+        public bool X => SyncAxis[0];
+
+        /// <summary>
+        /// When enabled, serialize the Y axis
+        /// </summary>
+        public bool Y => SyncAxis[1];
+
+        /// <summary>
+        /// When enabled, serialize the Z axis
+        /// </summary>
+        public bool Z => SyncAxis[2];
+
+        public bool[] SyncAxis;
+
+        /// <summary>
+        /// Constructor to preinitialize the x, y, and z values
+        /// </summary>
+        /// <param name="x">when <see cref="true"/> the x-axis will be serialized</param>
+        /// <param name="y">when <see cref="true"/> the y-axis will be serialized</param>
+        /// <param name="z">when <see cref="true"/> the z-axis will be serialized</param>
+        public HalfVector3AxisToSynchronize(bool x = true, bool y = true, bool z = true)
+        {
+            SyncAxis = new bool[3];
```

Comment:
This will allocate managed memory. Do we care about that? Could we make this a NativeArray instead, or just three bool values?

Reply:
Actually even better would be `FixedList32Bytes`, which gives the list interface in a fixed memory so it doesn't require any allocators or adding IDisposable. Slightly too large for three bools so it wouldn't have a length of 3 and you'd have to account for that... having just bool x, bool y, bool z would be the best option in terms of memory usage and allocation behavior.

Reply:
Another option would be a simple class like this:

```csharp
public struct Vector3T<T> where T: unmanaged
{
    // No other members, ensures array-style access works
    public T X;
    public T Y;
    public T Z;
    
    // Property, not value
    public int Length => 3;
    
    public unsafe T this[int index]
    {
        get 
        {
            fixed (FastBufferWriter* ptr = &this)
            {
                return ((T*)ptr)[index];
            }
        }
        set 
        {
            fixed (FastBufferWriter* ptr = &this)
            {
                ((T*)ptr)[index] = value;
            }
        }
    }
}
```

If you wanted, you could even give that an interface:

```csharp
public interface VectorNT<T>
{
    int Length { get; };
    T this[int index] { get; set };
}
```

Then you could make Vector2T, Vector3T, Vector4T, and pass them around using Length and array-access.

Reply:
of course... I will replace:
`FastBufferWriter* ptr` with `Vector3T<T>*`

But I get the general direction you were heading with that.

Reply:
Oh... yeah... oops. I wanted to test my syntax was correct so I just wrote some test code inside the class I happened to have open and forgot to adjust it.